pkg/network/telemetry/metric_group.go

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// MetricGroup provides a convenient constructor for a group with metrics
+// sharing the same namespace and group of tags.
+// Note the usage of this API is entirely optional; I'm only adding this here
+// to keep compatibility with some common patterns I've seen in the codebase.
+type MetricGroup struct {
+	mux        sync.Mutex
```

Comment:
can benefit from using `RWMutex` instead

Summary can lock for read, while other methods should lock for write

Reply:
I actually don't think that a `RWMutex` is a good fit here. The benefit is generally for multiple concurrent _readers_. See https://pkg.go.dev/sync#RWMutex
> The lock can be held by an arbitrary number of readers *or a single writer*.
In any case, I don't expect to see any contention happening here in a "real-world" usage of this